lib/components/AlbumScreen/track_list_tile.dart

```diff
@@ -633,51 +609,24 @@ class TrackListItemTile extends ConsumerWidget {
             (additionalBaseItemInfos[BaseItemDtoType.track] == AdditionalBaseItemInfoTypes.dateLastPlayed ||
                 additionalBaseItemInfos[BaseItemDtoType.track] == AdditionalBaseItemInfoTypes.playCount))
         ? AdditionalBaseItemInfoTypes.none
-        : additionalBaseItemInfos[BaseItemDtoType.track];
-
-    final additionalInfo = switch (additionalBaseItemInfo) {
-      AdditionalBaseItemInfoTypes.playCount => (
-        playCount: true,
-        releaseDate: false,
-        dateAdded: false,
-        dateLastPlayed: false,
-      ),
-      AdditionalBaseItemInfoTypes.dateReleased => (
-        playCount: false,
-        releaseDate: true,
-        dateAdded: false,
-        dateLastPlayed: false,
-      ),
-      AdditionalBaseItemInfoTypes.dateAdded => (
-        playCount: false,
-        releaseDate: false,
-        dateAdded: true,
-        dateLastPlayed: false,
-      ),
-      AdditionalBaseItemInfoTypes.dateLastPlayed => (
-        playCount: false,
-        releaseDate: false,
-        dateAdded: false,
-        dateLastPlayed: true,
-      ),
-      AdditionalBaseItemInfoTypes.none => (
-        playCount: false,
-        releaseDate: false,
-        dateAdded: false,
-        dateLastPlayed: false,
-      ),
-      _ => (
-        playCount: showPlayCountAdaptive,
-        releaseDate: showReleaseDateAdaptive,
-        dateAdded: showDateAddedAdaptive,
-        dateLastPlayed: showDateLastPlayedAdaptive,
-      ),
-    };
-
-    final showPlayCount = additionalInfo.playCount;
-    final showReleaseDate = additionalInfo.releaseDate;
-    final showDateAdded = additionalInfo.dateAdded;
-    final showDateLastPlayed = additionalInfo.dateLastPlayed;
+        : (additionalBaseItemInfos[BaseItemDtoType.track] ?? AdditionalBaseItemInfoTypes.adaptive);
+
+    bool showPlayCount =
+        (additionalBaseItemInfo == AdditionalBaseItemInfoTypes.playCount ||
+        (additionalBaseItemInfo == AdditionalBaseItemInfoTypes.adaptive &&
+            adaptiveAdditionalInfoSortBy == SortBy.playCount));
+    bool showReleaseDate =
+        (additionalBaseItemInfo == AdditionalBaseItemInfoTypes.dateReleased ||
+        (additionalBaseItemInfo == AdditionalBaseItemInfoTypes.adaptive &&
+            adaptiveAdditionalInfoSortBy == SortBy.premiereDate));
+    bool showDateAdded =
+        (additionalBaseItemInfo == AdditionalBaseItemInfoTypes.dateAdded ||
+        (additionalBaseItemInfo == AdditionalBaseItemInfoTypes.adaptive &&
+            adaptiveAdditionalInfoSortBy == SortBy.dateCreated));
+    bool showDateLastPlayed =
+        (additionalBaseItemInfo == AdditionalBaseItemInfoTypes.dateLastPlayed ||
+        (additionalBaseItemInfo == AdditionalBaseItemInfoTypes.adaptive &&
+            adaptiveAdditionalInfoSortBy == SortBy.datePlayed));
```

Comment:
I think this is a bit easier to read:
```suggestion
        : (additionalBaseItemInfos[BaseItemDtoType.track] ?? AdditionalBaseItemInfoTypes.adaptive);

    bool showPlayCount = additionalBaseItemInfo == AdditionalBaseItemInfoTypes.playCount:
    bool showReleaseDate = additionalBaseItemInfo == AdditionalBaseItemInfoTypes.dateReleased;
    bool showDateAdded = additionalBaseItemInfo == AdditionalBaseItemInfoTypes.dateAdded;
    bool showDateLastPlayed = additionalBaseItemInfo == AdditionalBaseItemInfoTypes.dateLastPlayed;;
    if (additionalBaseItemInfo == AddititionalBaseItemInfoType.adaptive) {
      showPlayCount = adaptiveAdditionalInfoSortBy == SortBy.playCount;
      showReleaseDate = adaptiveAdditionalInfoSortBy == SortBy.premiereDate;
      showDateAdded = adaptiveAdditionalInfoSortBy == SortBy.dateCreated;
      showDateLastPlayed = adaptiveAdditionalInfoSortBy == SortBy.datePlayed;
    }
```